packages/vscode-extension/src/commands/CreateFile.ts

```diff
@@ -0,0 +1,209 @@
+				),
+			};
+		}
+
+		return {
+			valid: true,
+		};
+	}
+
+	protected async execute(): Promise<void> {
+		const commandArgs = await this.getCommandArgs();
+		if (commandArgs === undefined) {
+			return;
+		}
+
+		const commandActionPromise = this.runSuiteCloudCommand(commandArgs);
+		const commandMessage = this.translationService.getMessage(COMMAND.TRIGGERED, this.vscodeCommandName);
+		const statusBarMessage: string = this.translationService.getMessage(CREATE_FILE.MESSAGES.CREATING_FILE);
+		this.messageService.showInformationMessage(commandMessage, statusBarMessage, commandActionPromise);
+
+		const actionResult = await commandActionPromise;
+		if (actionResult.isSuccess()) {
+			const createdFileUri: Uri = Uri.file(actionResult.data.path);
+			workspace.openTextDocument(createdFileUri).then(
+				(createdFile: TextDocument) => {
+					window.showTextDocument(createdFile);
+				},
+				(error: any) => {
+					this.messageService.showCommandError(error);
+				}
+			);
+		} else {
+			this.messageService.showCommandError();
+		}
+	}
+
+	private async getCommandArgs(): Promise<{ [key: string]: string | string[] } | undefined> {
+		const args: { [key: string]: string | string[] } = {
+			project: this.getProjectFolderPath(),
+		};
+
+		const selectedScriptType = await this.promptScriptTypeQuestion();
+		if (selectedScriptType === undefined) {
+			return;
+		} else {
+			args.type = selectedScriptType.id;
+		}
+
+		const selectedModules = await this.promptAddModulesQuestion();
+		if (selectedModules === undefined) {
+			return;
+		} else if (Array.isArray(selectedModules) && selectedModules.length > 0) {
+			args.module = selectedModules;
+		}
+
+		const selectedFolder = await this.promptFolderSelection();
+		if (selectedFolder === undefined) {
+			return;
+		}
+		const fileName = await this.promptFileNameInputBox(selectedFolder);
+		if (fileName === undefined) {
+			return;
+		}
+
+		args.path = path.join(selectedFolder, fileName);
+
+		return args;
+	}
+
+	private promptScriptTypeQuestion(): Thenable<SuiteScriptTypeItem | undefined> {
+		return window.showQuickPick(
+			SUITESCRIPT_TYPES.map((el) => <SuiteScriptTypeItem>{ label: el.name, id: el.id, name: el.name }),
+			{
+				placeHolder: this.translationService.getMessage(CREATE_FILE.QUESTIONS.CHOOSE_SUITESCRIPT_TYPE),
+				canPickMany: false,
+			}
+		);
+	}
+
+	private async promptAddModulesQuestion(): Promise<string[] | undefined> {
+		return window.showQuickPick(
+			SUITESCRIPT_MODULES.map((module) => module.id),
+			{
+				placeHolder: this.translationService.getMessage(CREATE_FILE.QUESTIONS.SELECT_SUITESCRIPT_MODULES),
+				canPickMany: true,
+			}
+		);
+	}
+
+	private async promptFolderSelection(): Promise<string | undefined> {
+		const folderChoices = this.getFolderChoices();
+
+		let fileToCheck = this.activeFile;
+
+		// action orignated from context menu
+		if (this.isFileSelected && fileToCheck && fs.existsSync(fileToCheck)) {
```

Comment:
Should we have something like a service for that can check if action originated from the context menu?

Reply:
I'm just using `this.isFileSelected` to know if the command was triggered from the context menu. The other conditions are just sanity checks.
Maybe we can rename `this.isFileSelected` -> `this.uriAvailable` or `this.fromContextMenu`. What do you think? I'm open to other suggestions.

Reply:
I was looking at the comment and I thought the whole if statement was for checking if the file was selected from the context menu. So yes, maybe rename this.isFileSelected to this.fromContextMenu if isFileSelected is just being used for checking if the file has been selected from the context menu.

Reply:
I'll check its usages, and rename accordingly.

Reply:
Renamed to `this.isSelectedFromContextMenu`